src/ray/core_worker/task_interface.cc

```diff
@@ -1,26 +1,155 @@
+                                      language, function.function_descriptor);
+
+  std::vector<ObjectID> execution_dependencies;
+  return core_worker_.raylet_client_->SubmitTask(execution_dependencies, spec);
 }
 
 Status CoreWorkerTaskInterface::CreateActor(
     const RayFunction &function, const std::vector<TaskArg> &args,
-    const ActorCreationOptions &actor_creation_options, ActorHandle *actor_handle) {
-  return Status::OK();
+    const ActorCreationOptions &actor_creation_options,
+    std::unique_ptr<ActorHandle> *actor_handle) {
+  auto &context = core_worker_.worker_context_;
+  auto next_task_index = context.GetNextTaskIndex();
+  const auto task_id = GenerateTaskId(context.GetCurrentDriverID(),
+                                      context.GetCurrentTaskID(), next_task_index);
+
+  std::vector<ObjectID> return_ids;
+  return_ids.push_back(ObjectID::ForTaskReturn(task_id, 1));
+  ActorID actor_creation_id = ActorID::FromBinary(return_ids[0].Binary());
+
+  *actor_handle = std::unique_ptr<ActorHandle>(
+      new ActorHandle(actor_creation_id, ActorHandleID::Nil()));
+  (*actor_handle)->IncreaseTaskCounter();
+  (*actor_handle)->SetActorCursor(return_ids[0]);
+
+  std::vector<std::shared_ptr<raylet::TaskArgument>> task_arguments;
+  for (auto &arg : args) {
+    if (arg.IsPassedByReference()) {
+      std::vector<ObjectID> references{arg.GetReference()};
+      task_arguments.push_back(
+          std::make_shared<raylet::TaskArgumentByReference>(references));
+    } else {
+      auto data = arg.GetValue();
+      task_arguments.push_back(
+          std::make_shared<raylet::TaskArgumentByValue>(data->Data(), data->Size()));
+    }
+  }
+
+  // Note that the caller is supposed to specify required placement resources
+  // correctly via actor_creation_options.resources.
+  ::Language language = (function.language == ray::WorkerLanguage::JAVA)
+                            ? (::Language::JAVA)
+                            : (::Language::PYTHON);
+
+  ray::raylet::TaskSpecification spec(
+      context.GetCurrentDriverID(), context.GetCurrentTaskID(), next_task_index,
+      actor_creation_id, ObjectID::Nil(), actor_creation_options.max_reconstructions,
+      ActorID::Nil(), ActorHandleID::Nil(), 0, {}, task_arguments, 1,
+      actor_creation_options.resources, actor_creation_options.resources, language,
+      function.function_descriptor);
+
+  std::vector<ObjectID> execution_dependencies;
+  return core_worker_.raylet_client_->SubmitTask(execution_dependencies, spec);
 }
 
 Status CoreWorkerTaskInterface::SubmitActorTask(ActorHandle &actor_handle,
                                                 const RayFunction &function,
                                                 const std::vector<TaskArg> &args,
-                                                const TaskOptions &task_options,
+                                                const TaskOptions &call_options,
```

Comment:
Was this change  unintentional? I didn't change the header file accordingly.

Reply:
ok, updated.